src/interface/state_descriptor.hpp

```diff
@@ -245,7 +245,34 @@ class StateDescriptor {
   Dictionary<Dictionary<Metadata>> swarmValueMetadataMap_;
 };
 
-using Packages_t = Dictionary<std::shared_ptr<StateDescriptor>>;
+class Packages_t {
+ public:
+  Packages_t() = default;
+  bool Add(const std::shared_ptr<StateDescriptor> &package) {
```

Comment:
Yeah, I'd say make it an error to call `Add` with the same name multiple times. I can't imagine that overwriting is ever the intended behavior. If people really want it, you could add a separate `AddOrOverwrite` method later.

Reply:
I think I agree. And I've made the change.